### DISCUSSION TITLE
Nokogiri adds extra `meta http-equiv="Content-Type"` to xhtml?

### DISCUSSION THREAD

**Discussion by jrochkind** [sparklemotion/nokogiri#3302] [General Q&A] [answered]
nokogiri (1.16.7 arm64-darwin)

If I have an XML document that `<!DOCTYPE` identifying it as XHTML, the output has an extra meta http-equiv content-type tag added to the `<head>` section. 

This meta tag isn't findable inside the visible nokogiri in-memory struture, but appears on serialization. 

Is this expected? Is there any way to turn this off or remove it? Thanks for any advice!

```ruby
require 'nokogiri'

input = <<~EOS
<?xml version="1.0"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
 <head>
 </head>
 <body>
</body>
</html>
EOS


noko = Nokogiri::XML(input)

puts noko.to_xml
```

```xml
<?xml version="1.0"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
 <head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
 </head>
 <body>
</body>
</html>
```

**Comment by flavorjones** [comment, marked as answer]
@jrochkind This behavior is from libxml2, originally added in gnome/libxml2@d5c2f92d (in 2002!).

As far as I can tell, this is here because of the [Character Encoding](https://www.w3.org/TR/xhtml1/guidelines.html#C_9) section of the XHTML1 spec, which to my reading is a bit ambiguous.

There's no way to turn this off, unfortunately.

Can I ask why this is an issue for you? I'd love to more deeply understand what you're trying to do.

**Reply by jrochkind** [reply]
Thanks!  Very odd, so libxml decided XHTML required it? Okay then. 

I am trying to produce XML output that semantically matches a target sample output I have as a requirement; in this case for "HOCR" format OCR text positioning information, which is a standard from the era of XHTML.   And I am trying to transform an input document in certain ways to get that output, and was writing tests for it being transformed only in the ways I expected, not adding extra `meta` tags. 

But I can't see the extra meta tag hurting (other than wasting some bytes).  I guess I can just change my requirements to say "oh and also it can add a meta http-equiv content-type , fine".  Or I can use a regexp to strip it out of the serialized output I guess. 

Not a big deal, just unexpected behavior that I wanted to report and check in on! Thank you!

**Reply by flavorjones** [reply]
OK! I'm glad you can work around it.
